assets/js/modules/analytics-4/datastore/properties.js

```diff
@@ -95,22 +94,16 @@ const fetchGetPropertiesStore = createFetchStore( {
 			}
 		);
 	},
-	reducerCallback( state, properties, { accountID } ) {
-		return {
-			...state,
-			properties: {
-				...state.properties,
-				[ accountID ]: properties,
-			},
-			propertiesByID: properties.reduce(
-				( accum, property ) => ( {
-					...accum,
-					[ property._id ]: property,
-				} ),
-				state.propertiesByID || {}
-			),
-		};
-	},
+	reducerCallback: createReducer( ( state, properties, { accountID } ) => {
+		state.properties = state.properties || {};
+		state.propertiesByID = state.propertiesByID || {};
+
+		state.properties[ accountID ] = properties;
+
+		for ( const property of properties ) {
```

Comment:
Please restore the previous logic, where `state.propertiesByID` is completely overwritten by the return value of the `properties.reduce()` call. As it stands, it's now possible for `state.propertiesByID` to retain some old values when it shouldn't do.